build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/JavaPaths.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using Xamarin.Android.Build.Utilities;
+
+namespace Xamarin.Android.BuildTools.PrepTasks
+{
+	public class JavaPaths : Task
+	{
+		public string AndroidNdkPath { get; set; }
+
+		public string AndroidSdkPath { get; set; }
+
+		public string JavaSdkPath { get; set; }
+
+		[Output]
+		public string JavaC { get; set; }
+
+		[Output]
+		public string Jar { get; set; }
+
+		public override bool Execute ()
+		{
+			LogMessages ();
+
+			AndroidLogger.Error += ErrorHandler;
+			AndroidLogger.Warning += WarningHandler;
+			AndroidLogger.Info += InfoHandler;
+			try {
+				AndroidSdk.Refresh (AndroidSdkPath, AndroidNdkPath, JavaSdkPath);
+
+				var javaSdkPath = AndroidSdk.JavaSdkPath;
+				if (string.IsNullOrEmpty (javaSdkPath)) {
+					Log.LogError ("JavaSdkPath is blank");
+					return false;
+				}
+
+				Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidSdk.JavaSdkPath)}: {javaSdkPath}");
```

Comment:
Current convention (e.g. from `<BuildApk/>`) is that output values should be logged with `[Output]`:

```csharp
Log.LogMessage (MessageImportance.Low, $"  {nameof (JavaC)}: {JavaC}");
Log.LogMessage (MessageImportance.Low, $"  {nameof (Jar)}: {Jar}");
```

I realize that `javaSdkPath` isn't an output variable, but it can also be inferred from the `JavaC` path.